src/main/java/springboot/onlinebookstore/controller/AuthenticationController.java

```diff
@@ -0,0 +1,39 @@
+package springboot.onlinebookstore.controller;
+
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import springboot.onlinebookstore.dto.user.UserLoginRequestDto;
+import springboot.onlinebookstore.dto.user.UserLoginResponseDto;
+import springboot.onlinebookstore.dto.user.UserRegistrationRequestDto;
+import springboot.onlinebookstore.dto.user.UserResponseDto;
+import springboot.onlinebookstore.exception.RegistrationException;
+import springboot.onlinebookstore.security.AuthenticationService;
+import springboot.onlinebookstore.service.UserService;
+
+@Slf4j
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/auth")
+public class AuthenticationController {
+    private final UserService userService;
+    private final AuthenticationService authenticationService;
+
```

Comment:
you are not asked to add authentication controller at this HW. 

Pls make PR with only relevant for this hw code.

Reply:
Sorry, I didn't understand the comment. The Authentication Controller was completed earlier as part of work on user registration and login capabilities. I simply continued developing the task based on the current application.